certbot-apache/certbot_apache/augeasparser.py

```diff
@@ -253,6 +289,72 @@ def _aug_get_block_name(self, path):
         name = name.split("[")[0]
         return name
 
+    def _aug_resolve_child_position(self, name, position):
```

Comment:
Will `self.metadata["augeaspath"]` ever already have a trailing slash? If it's possible, I think we should check for this case and make sure the path we construct doesn't contain two consecutive slashes as Augeas' behavior was different when testing this case locally.

Reply:
It _should_ not. We however used that code in `ApacheParser` and I decided to bring it over to ensure consistency. Do you think it'd be better to just raise an exception instead to catch the occurances (if any) early?

Reply:
Are you referring to specific code in `ApacheParser` or just that we make the assumption that the Augeas path won't end in a trailing slash in most places?

I personally think writing an assertion in `AugeasParserNode.__init__` or `AugeasDirectiveNode.__init__` to cause the code to fail fast if the given `augeaspath` ends in a trailing slash is a good idea.

Reply:
It was based on the code that checks the path in [current parser](https://github.com/certbot/certbot/blob/master/certbot-apache/certbot_apache/parser.py#L649-L650). According to git log, I myself wrote those lines, and just half a year ago. I can't remember the reason for the check anymore though. I think it was added, because we do a lot of manual Augeas path string manipulation in the current code.

In the ParserNode implementation, the path creation is managed better, and we shouldn't have the same issue. I'll a add the check to `__init__` nevertheless.

Reply:
Sounds good to me!

We could alternatively just fix the problem in `__init__` and remove the trailing slash rather than erroring if you prefer, but I personally think both of those approaches are better than possibly letting the problem go unchecked and erroring much later in execution.